transport/response.go

```diff
@@ -24,5 +24,6 @@ import "io"
 
 // Response is the low level response representation.
 type Response struct {
-	Body io.ReadCloser
+	Headers map[string]string
+	Body    io.ReadCloser
```

Comment:
I suspect that this is not sufficiently generic to straddle TChannel and HTTP. Do headers conflate transport headers and application headers? Is it the transport’s responsibility to divide these between transport and application when preparing the response? Likewise for request.